src/particle/corsika/corsika7id.py

```diff
@@ -0,0 +1,168 @@
+# Copyright (c) 2018-2022, Eduardo Rodrigues and Henry Schreiner.
+#
+# Distributed under the 3-clause BSD license, see accompanying file LICENSE
+# or https://github.com/scikit-hep/particle for details.
+
+"""
+Class representing a Corsika7 ID.
+
+Note
+----
+Corsika8 uses Geant3 Particle IDs.
+"""
+
+
+from __future__ import annotations
+
+import csv
+from typing import TypeVar
+
+from .. import data
+from ..exceptions import MatchingIDNotFound
+from ..pdgid import PDGID
+
+Self = TypeVar("Self", bound="Corsika7ID")
+
+
+with data.basepath.joinpath("pdgid_to_corsika7id.csv").open() as _f:
+    _bimap = {
+        int(v["CORSIKA7ID"]): int(v["PDGID"])
+        for v in csv.DictReader(line for line in _f if not line.startswith("#"))
+    }
+
+# Some Corsika ID's are not really particles
+_non_particles = {
+    71: "η → γγ",
+    72: "η → 3π◦",
+    73: "η → π+π−π◦",
+    74: "η → π+π−γ",
+    75: "μ+ add. info.",
+    76: "μ− add. info.",
+    85: "decaying μ+ at start",
+    86: "decaying μ− at start",
+    95: "decaying μ+ at end",
+    96: "decaying μ− at end",
+}
+
+
+class Corsika7ID(int):
+    """
+    Holds a Corsika7 ID.
+
+    Examples
+    --------
+    >>> cid = Corsika7ID(6)
+
+    >>> from particle import Particle
+    >>> p = Particle.from_pdgid(cid.to_pdgid())
+
+    >>> (p,) = Particle.finditer(pdgid=cid.to_pdgid())
+    >>> p.name
+    'mu-'
+    """
```

Comment:
Any reason not to have the line `__slots__ = ()  # Keep PythiaID a slots based class` as in similar classes?

Reply:
No, none at all. Never encountered `__slots__` and honestly, I just didn't understand it. Seems I still have a long way to go until python expertise ;) I'll add it back :)
```suggestion
    """
    
    __slots__ = ()  # Keep Corsika7ID a slots based class
```